src/Cli/dotnet/CommonOptions.cs

```diff
@@ -100,6 +116,57 @@ public static bool VerbosityIsDetailedOrDiagnostic(this VerbosityOptions verbosi
                 verbosity.Equals(VerbosityOptions.d) ||
                 verbosity.Equals(VerbosityOptions.detailed);
         }
+
+        internal static IEnumerable<string> ResolveArchOptionToRuntimeIdentifier(string arg, ParseResult parseResult)
+        {
+            if (parseResult.BothArchAndOsOptionsSpecified())
+            {
+                return Array.Empty<string>();
+            }
+
+            return ResolveRidShorthandOptions(null, arg);
+        }
+
+        internal static IEnumerable<string> ResolveOsOptionToRuntimeIdentifier(string arg, ParseResult parseResult)
+        {
+            if (parseResult.BothArchAndOsOptionsSpecified())
+            {
+                return ResolveRidShorthandOptions(arg, parseResult.ValueForOption<string>(CommonOptions.ArchitectureOption().Aliases.First()));
+            }
+
+            return ResolveRidShorthandOptions(arg, null);
+        }
+
+        private static IEnumerable<string> ResolveRidShorthandOptions(string os, string arch)
+        {
+            var currentRid = GetCurrentRuntimeId();
+            os = string.IsNullOrEmpty(os) ? GetOsFromRid(currentRid) : os;
+            arch = string.IsNullOrEmpty(arch) ? GetArchFromRid(currentRid) : arch;
+            var properties = new string[] { $"-property:RuntimeIdentifier={os}-{arch}" };
+            return properties;
+        }
+
+        private static string GetCurrentRuntimeId()
+        {
+            var dotnetRootPath = Path.GetDirectoryName(Environment.ProcessPath);
+            dotnetRootPath = dotnetRootPath.Contains("dotnet") ? dotnetRootPath : Path.Combine(dotnetRootPath, "dotnet");
```

Comment:
I'm not following what this line is doing.  Under what circumstances would you need to append "dotnet" to the path?

Reply:
In testing. Adding this line allows us to pick up the rid file when running unit tests, without it we fail and can't test.

Reply:
Why - do we need to modify the tests to better mimic the actual runtime environment?

Reply:
Environment.ProcessPath gives us the location of dotnet.exe when we're really running commands, but in tests it gives us testhost.exe (which is on dir above the location of dotnet.exe). The same difference is what causes Product.Version to be an empty string when testing.

Reply:
Good point - can we make it more specific then?
Basically switch the code around instead of looking for "dotnet" look for "testhost" and then do "the other thing". It would make the code cleaner in my opinion. And I would still add a comment describing how it looks under test (basically what you wrote above in your reply here).

Reply:
Sure, just pushed a commit with that change.